Update Backlinks docs

marcusolsson opened the pull request:
Update for the Backlinks plugin docs. I took the liberty of using mermaid here, let me know if you'd prefer not to.

ericaxu commented:
Looks good!

The original doc is outdated, do you think you can fit in "filtering backlink" and the "backlink in document" option in ikts plugin options.

marcusolsson commented:
- Added a "backlink in document" as a tip rather than as reference documentation. 
- Added a reference to Search since they work similarly.

Let me know what you think!

ericaxu commented:
LGTM!
